### PR TITLE
Add BDSP underground tables and data

### PR DESCRIPTION
**Pull request by SteveCookTU**

### CODE REVIEW

File: Gen8/bdsp.py [hunk 1]
```diff
@@ -10,6 +10,19 @@ def encounters():
...
+    UG_ENCOUNT_05 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_05.json"
+    UG_ENCOUNT_06 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_06.json"
+    UG_ENCOUNT_07 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_07.json"
+    UG_ENCOUNT_08 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_08.json"
+    UG_ENCOUNT_09 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_09.json"
+    UG_ENCOUNT_10 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_10.json"
+    UG_ENCOUNT_11 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_11.json"
+    UG_ENCOUNT_12 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_12.json"
+    UG_ENCOUNT_20 = f"{SCRIPT_FOLDER}/bdsp/UgEncount_20.json"
+    UG_SPECIAL_POKEMON = f"{SCRIPT_FOLDER}/bdsp/UgSpecialPokemon.json"
```
Comment (Admiral-Fish) [inline Gen8/bdsp.py]: General comment here reviewing what was done in gen 4, this should all be moved to a underground function

File: Gen8/bdsp.py [hunk 2]
```diff
@@ -190,15 +244,174 @@ def encounters():
             p += entry["minlv"].to_bytes(1, "little")
             p += entry["monsNo"].to_bytes(2, "little")
 
+    for pokemon in ug_encount_02:
+        ug_encount += (2).to_bytes(1, "little")
```
Comment (Admiral-Fish) [inline Gen8/bdsp.py]: General comment here: if the number you are putting is a constant just use the byte equivalent

ex: b"\x02"

File: Gen8/bdsp.py [hunk 3]
```diff
@@ -190,15 +244,174 @@ def encounters():
...
+        f.write(ug_special_pokemon)
+
     with open("bdsp_en.txt", "w+", encoding="utf-8") as f:
         map_names.sort(key=lambda x: x[0])
         for i, (num, name) in enumerate(map_names):
             f.write(f"{num},{name}")
             if i != len(map_names) - 1:
                 f.write("\n")
+
+def ug_extra_data():
```
Comment (Admiral-Fish) [inline Gen8/bdsp.py]: Rename to `underground` and have all the underground related stuff here

**Review by Admiral-Fish — CHANGES_REQUESTED** [review]
General question: are all the json dumped directly from the game without modification?

**Comment by SteveCookTU** [comment]
Yes, all files are directly from the game without modification

File: Gen8/bdsp.py [hunk 4]
```diff
@@ -202,3 +202,208 @@ def encounters():
...
+
+    ug_encount = bytes()
+    ug_special_pokemon = bytes()
+    ug_rand_mark = bytes()
+    tamago_waza_table = bytes()
+    tamago_waza_ignore_table = bytes()
+    ug_pokemon_data = bytes()
+
+    for pokemon in ug_encount_02:
+        ug_encount += b"\x02"
```
Comment (Admiral-Fish) [inline Gen8/bdsp.py]: General question for all of the ug_encount_XX loops:
1. Is this value even necessary? What do you use it for? Can we not assume fixed offsets assuming it is some sort of clarifying value to know where we are in the data?
2. Assuming it is necessary, can it be moved out of the loop? It gets duplicated for every pokemon entry

Comment (SteveCookTU) [reply Gen8/bdsp.py]: The amount of pokemon in each table isn't the same so the other way around would be to replace this value with the amount of entries so we know how many are in each and then can run based on indices from there. The reason it is there for each is to identify which entry belongs to which table in the long run. 

The number of entries is fixed per but that would mean hard coding the number for each section within finder (which personally I'm not a fan of hard coded values like that).

**Comment by Admiral-Fish** [comment]
This was included via 5f09ccc

**Pull request closed without merging**